components/bsx/Create/CreateCollection.vue

```diff
@@ -62,143 +57,153 @@ import shouldUpdate from '@/utils/shouldUpdate'
+}
 
-  @Watch('accountId', { immediate: true })
-  async onAccountIdChange(val: string, oldVal: string) {
-    if (shouldUpdate(val, oldVal)) {
-      this.feesToken = await getFeesToken()
-    }
+const balanceNotEnoughMessage = computed(() => {
+  if (balanceNotEnough.value) {
+    return t('tooltip.notEnoughBalance')
   }
+  return ''
+})
 
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const classDeposit = getclassDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.collectionDeposit = (classDeposit + metadataDeposit).toString()
-    })
-  }
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
+watch(accountId, async (newVal, oldVal) => {
+  if (shouldUpdate(newVal, oldVal)) {
+    feesToken.value = await getFeesToken()
   }
+})
 
-  get depositOfToken() {
-    return getDeposit(this.feesToken, parseFloat(this.collectionDeposit))
-  }
+const created = async () => {
+  onApiConnect(apiUrl.value, (api) => {
+    const classDeposit = getclassDeposit(api)
+    const metadataDeposit = getMetadataDeposit(api)
+    collectionDeposit.value = (classDeposit + metadataDeposit).toString()
+  })
+}
 
-  protected async tryToEstimateTx(): Promise<string> {
-    const api = await ApiFactory.useApiInstance(this.apiUrl)
-    const cb = api.tx.utility.batchAll
-    const metadata = dummyIpfsCid()
-    const randomId = 0
-    const args = [createArgs(randomId, metadata)]
-    return estimate(this.accountId, cb, args)
+const balanceOfToken = computed(() => {
+  return getBalance(feesToken.value)
+})
+
+const depositOfToken = computed(() => {
+  return getDeposit(feesToken.value, parseFloat(collectionDeposit.value))
+})
+
+const tryToEstimateTx = async () => {
+  const api = await ApiFactory.useApiInstance(apiUrl.value)
+  const cb = api.tx.utility.batchAll
+  const metadata = dummyIpfsCid()
+  const randomId = 0
+  const args = [createArgs(randomId, metadata)]
+  return estimate(accountId.value, cb, args)
+}
+
+const checkBalanceBeforeTx = async () => {
+  const estimated = await tryToEstimateTx()
+  const deposit = collectionDeposit.value
+  const hasTokens = hasEnoughToken(balance.value, estimated, deposit)
+  $consola.log('hasTokens', hasTokens)
+  if (!hasTokens) {
+    throw new Error(
+      `Not enough tokens: Currently have ${formatBalance(
+        balance.value,
+        decimals.value,
+        unit.value
+      )} tokens`
+    )
   }
+}
 
-  protected async checkBalanceBeforeTx(): Promise<void> {
-    const estimated = await this.tryToEstimateTx()
-    const deposit = this.collectionDeposit
-    const hasTokens = hasEnoughToken(this.balance, estimated, deposit)
-    this.$consola.log('hasTokens', hasTokens)
-    if (!hasTokens) {
-      throw new Error(
-        `Not enough tokens: Currently have ${formatBalance(
-          this.balance,
-          this.decimals,
-          this.unit
-        )} tokens`
-      )
-    }
+const submit = async () => {
```

Comment:
I believe this should be improved, watchers shouldn't be placed inside this function. This registers the watchers every time submit is clicked.

Reply:
Yeah, I left everything in before getting feedback. I will remove the unused methods and improve the `submit` method.